src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -169,6 +173,53 @@ def pex_requirements_for_default_lockfile(cls):
             resolve_name=cls.options_scope,
         )
 
+    @classproperty
+    def help_extended(cls) -> str:
+        base_help = cls.help if isinstance(cls.help, str) else cls.help()
+        if cls.default_lockfile_resource is None:
+            return base_help
+
+        lockfile = cls.pex_requirements_for_default_lockfile()
+        parts = urlparse(lockfile.url)
+        # urlparse retains the leading / in URLs with a netloc.
+        lockfile_path = parts.path[1:] if parts.path.startswith("/") else parts.path
+        if parts.scheme in {"", "file"}:
+            with open(lockfile_path, "rb") as fp:
+                lock_bytes = fp.read()
+        elif parts.scheme == "resource":
+            _fc = FileContent(
+                lockfile_path,
+                # The "netloc" in our made-up "resource://" scheme is the package.
+                importlib.resources.read_binary(parts.netloc, lockfile_path),
+            )
+            lockfile_path, lock_bytes = (_fc.path, _fc.content)
```

Comment:
We might be able to write:

```suggestion
            # The "netloc" in our made-up "resource://" scheme is the package.
            lock_bytes = importlib.resources.read_binary(parts.netloc, lockfile_path)
```